heat/core/arithmetics.py

```diff
@@ -542,6 +586,38 @@ def floordiv(t1: Union[DNDarray, float], t2: Union[DNDarray, float]) -> DNDarray
 """Alias for :py:func:`floordiv`"""
 
 
+def hypot(
+    a: DNDarray, b: DNDarray, /, out: Optional[DNDarray] = None, *, where: Optional[DNDarray] = True
+) -> DNDarray:
+    """
+    Given the 'legs' of a right triangle, return its hypotenuse.
+
+    Parameters
+    ----------
+    a:   DNDarray
+         The first input array
+    b:   DNDarray
+         the second input tensor
+    out: DNDarray, optional
+        The output array. It must have a shape that the inputs broadcast to and matching split axis.
+        If not provided, a freshly allocated array is returned.
+    where: DNDarray, optional
+        Condition to broadcast over the inputs. At locations where the condition is True, the `out` array
+        will be set to the divided value. Elsewhere, the `out` array will retain its original value. If
+        an uninitialized `out` array is created via the default `out=None`, locations within it where the
+        condition is False will remain uninitialized. If distributed, the split axis (after broadcasting
+        if required) must match that of the `out` array.
+
+    Examples
+    --------
+    >>> a = a=ht.array([2.])
+    >>> b = b=ht.array([1.,3.,3.])
+    >>> ht.hypot(a,b)
+    DNDarray([2.2361, 3.6056, 3.6056], dtype=ht.float32, device=cpu:0, split=None)
+    """
+    return _operations.__binary_op(torch.hypot, a, b, out, where)
```

Comment:
Actually, I am not really sure what the typical use-case for this function is, but would it make sense to add a check whether the inputs are positive?

Reply:
One example is calculating the length of a 2d vector. I will add a sentence with the equivalent formula for better understanding.